### PR TITLE
[django] Generalize url/key logic

### PR DESCRIPTION
**Pull request by waleko**
Removed the logics which were duplicated
in the others modules and redefined it here to enable reusability and ease debugging.

Related to openwisp/openwisp-utils#40

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -13,8 +13,11 @@ branches:
     - master
 
 install:
-  - # TODO: this is temporary, remove it when django-freeradius 0.1 is released
+  # TODO: this is temporary, remove it when django-freeradius 0.1 is released
   - pip install https://github.com/openwisp/django-freeradius/tarball/master
+  # Temporary: remove when openwisp-utils#71 is merged
+  - pip install "https://github.com/waleko/openwisp-utils/tarball/issues/40" --upgrade
```
Comment (nemesisdesign) [inline .travis.yml]: I released openwisp-utils 0.4.1 which contains your patch openwisp/openwisp-utils#71.
You can now remove this temporary addition.

File: requirements-test.txt [hunk 2]
```diff
@@ -1,7 +1,7 @@
 coveralls
 responses
 freezegun
-openwisp-utils[qa]>=0.3.2
+openwisp-utils[qa]>=0.4.0
```
Comment (nemesisdesign) [inline requirements-test.txt]: set to 0.4.1

File: requirements.txt [hunk 3]
```diff
@@ -1,3 +1,4 @@
 django-freeradius<0.2.0
 openwisp-users>=0.1.12,<0.2
 django-rest-auth>=0.9.3,<0.12.0
+openwisp-utils[qa]>=0.4.1
```
Comment (nemesisdesign) [inline requirements.txt]: here we don't need it because it's defined in django-freeradius, I already updated it there.

**Comment by waleko** [comment]
Got it

**Pull request merged**